### ISSUE TITLE
Reproducing the result of CONLL2014(test) using XLNet

### ISSUE THREAD

**Issue by MichaelCaohn** [grammarly/gector#44]
Hi，

Thank you very much for the great work. I have been trying to reproduce the result for CONLL2014. I have obtained the following results for the stage-1, stage-2, stage-3 and the Inf. Tweaks stage:
![image](https://user-images.githubusercontent.com/22003919/89382872-17717380-d72e-11ea-961f-f7483ca9e99b.png)

I observe that for the previous three stages, the difference between the reproduced result and the original result in the paper is not very large. However, for the Inf. Tweak stage the difference is quite large. I am using the following scripts: 

"python predict.py --model_path=model_weights/xlnet-2-gpu/stage-3/best.th --vocab_path=data/output_vo
cabulary --input_file=COLL14/conll14st-test.original --output_file=output/xlnet-stage-3.txt --transformer_model=xlnet --special_tokens_fix=0 --additional_confidence=0.35 --min_error_probability=0.66"

May I know how to solve this problem?

Thank you in advance.

**Comment by skurzhanskyi** [comment]
Hi @MichaelCaohn 
It's great news you having close results in the first 3 stages, but I see your model more precise but has lower recall. 
Parameters like `additional_confidence` and `min_error_probability` should be tuned for each particular model. If I'm not mistaken, we used BEA-dev for this purpose.

**Comment by MichaelCaohn** [comment]
So, additional_confidence=0.35 and min_error_probability=0.66 are only for the XLNet model you have uploaded in the repository and for BEA-dev set only? We have to tune these two parameters again for each of our self-trained model?

**Comment by skurzhanskyi** [comment]
It's not only for BEA-dev but in general. 
That's why they're called Inference Tweaks. They are used to change precision/recall tradeoff.
So it's logical that you need to have them different for different models.

**Comment by skurzhanskyi** [comment]
If you want to reproduce the scores reported in the paper there're links to the pre-trained model in the README.

**Comment by MichaelCaohn** [comment]
Thank you very much, 

May I know for searching these two hyperparameters' value. Did you just use the predict.py code to search it manually on the BEA-dev set?

**Comment by komelianchuk** [comment]
Yes, we did just a manual search on the BEA-dev set for those two.

I'm wondering why your stage3 results are that much different from ours (in terms of precision/recall balance).
In our case, stage3 training finished after 6 epoch and 4th (which is model_state_epoch_3.th) was the best one.
```
{
  "best_epoch": 3,
  "peak_cpu_memory_MB": 4149.808,
  "peak_gpu_0_memory_MB": 3716,
  "training_duration": "0:20:34.221210",
  "training_start_epoch": 0,
  "training_epochs": 5,
  "epoch": 5,
  "training_accuracy": 0.9473452462514038,
  "training_loss": 0.2823439410289204,
  "training_cpu_memory_MB": 4149.808,
  "training_gpu_0_memory_MB": 3716,
  "validation_accuracy": 0.9456371472716922,
  "validation_loss": 0.3225328190760179,
  "best_validation_accuracy": 0.94555582662438,
  "best_validation_loss": 0.31785173307765613
}
```
And what about your training stats?

**Comment by MichaelCaohn** [comment]
Hi, May I know where to find those training stats?

**Comment by komelianchuk** [comment]
Sure.
It should be stored in the serialization directory (model_directory).
You need to find files `metrics_epoch_*.json`

**Comment by MichaelCaohn** [comment]
Thank you very much,

I have found the training stats for stage-3:
It trained for 4 epochs, and the best model is model 2(model_state_epoch_1.th)

{
  "best_epoch": 1,
  "peak_cpu_memory_MB": 4498.456,
  "peak_gpu_0_memory_MB": 3530,
  "training_duration": "0:02:11.316700",
  "training_start_epoch": 0,
  "training_epochs": 3,
  "epoch": 3,
  "training_accuracy": 0.9401963464079085,
  "training_loss": 0.3874587427605601,
  "training_cpu_memory_MB": 4498.456,
  "training_gpu_0_memory_MB": 3530,
  "validation_accuracy": 0.9428134556574923,
  "validation_loss": 0.4290752559900284,
  "best_validation_accuracy": 0.9412844036697248,
  "best_validation_loss": 0.42840154469013214
}

**Comment by komelianchuk** [comment]
The difference in the loss is too big.
What were your training parameters for xlnet during stage3?

**Comment by MichaelCaohn** [comment]
Hi, 

my training hyperparameters in xlnet for stage3 is：
”python3 train.py --train_set=Stage-3-data/train-tagged.txt --dev_set=Stage-3-data/dev-tagged.txt --model_dir=model_weights/xlnet-2-gpu/stage-3 --cold_steps_count=0 --accumulation_size=2 --updates_per_epoch=0 --tn_prob=1 --tp_prob=1 --transformer_model=xlnet --special_tokens_fix=0 --tune_bert=1 --skip_correct=1 --skip_complex=0 --n_epoch=20 --patience=3 --max_len=50 --batch_size=64 --tag_strategy=keep_one --cold_steps_count=0 --cold_lr=1e-3 --lr=1e-5 --predictor_dropout=0.0 --lowercase_tokens=0 --pieces_per_token=5 --vocab_path=data/output_vocabulary --label_smoothing=0.0 --pretrain_folder=model_weights/xlnet-2-gpu/stage-2 --pretrain=best“

**Comment by komelianchuk** [comment]
The parameters are correct.
I think here you have the same issue with #45.
I would suggest to train a model without early stopping (patience=0) and after each stage chose a checkpoint with the most similar scores on BEA19-dev to ours.

**Comment by MichaelCaohn** [comment]
Thank you very much.
